massa-execution-worker/src/execution.rs

```diff
@@ -248,7 +248,56 @@ impl ExecutionState {
         if slot > max_slot {
             panic!("cannot execute at a slot beyond active cursor + 1");
         }
+    }
+
+    /// Computes the index of a given slot in the active history
+    fn get_active_index(&self, slot: Slot) -> Option<usize> {
+        let current = self.active_cursor.period * (self.config.thread_count as u64)
+            + (self.active_cursor.thread as u64);
+        let asked = slot.period * (self.config.thread_count as u64) + (slot.thread as u64);
+        current.checked_sub(asked).and_then(|v| v.try_into().ok())
+    }
+
+    /// Lazily query (from end to beginning) the active balance of an address at a given slot.
+    /// Returns None if the address balance could not be determined from the active history.
+    ///
+    /// NOTE: temporary, needs to be done in the speculative ledger
+    pub fn fetch_active_history_balance(
+        &self,
+        slot: Slot,
+        addr: &Address,
+    ) -> HistorySearchResult<Amount> {
+        self.verify_active_slot(slot);
```

Comment:
let's make sure the panic in there cannot be triggered by someone asking for the balance at a block in the future through the api

Reply:
This function is currently never used, I left it there as a blueprint for my future implementation of speculative ledger lazy sub-queries. No chance of that happening right now, I noted it and will make sure it cannot happen when it is actually used by the API.

Reply:
okay